test/Quaternion.jl

```diff
@@ -132,25 +208,71 @@ Base.:(/)(a::MyReal, b::Real) = a.val / b
         end
     end
 
-    @testset "isreal" begin end
-
-    @testset "iszero" begin end
-
-    @testset "isone" begin end
-
-    @testset "isfinite" begin end
+    @testset "isreal" begin
+        @test isreal(Quaternion(1, 0, 0, 0))
+        @test !isreal(Quaternion(2, 1, 0, 0))
+        @test !isreal(Quaternion(2, 0, 1, 0))
+        @test !isreal(Quaternion(2, 0, 0, 1))
+    end
 
-    @testset "isinf" begin end
+    @testset "iszero" begin
+        @test iszero(Quaternion(0.0, 0.0, 0.0, 0.0))
+        @test !iszero(Quaternion(1.0, 0.0, 0.0, 0.0))
+        @test !iszero(Quaternion(0.0, 1.0, 0.0, 0.0))
+        @test !iszero(Quaternion(0.0, 0.0, 1.0, 0.0))
+        @test !iszero(Quaternion(0.0, 0.0, 0.0, 1.0))
+        @test !iszero(Quaternion(0.0, 0.0, 0.0, 0.0, true))
+    end
 
-    @testset "isnan" begin end
+    @testset "isone" begin
+        @test isone(Quaternion(1))
+        @test !isone(Quaternion(-1))
+        @test !isone(Quaternion(0, 1, 0, 0))
+        @test !isone(Quaternion(1, 1, 0, 0))
+        @test !isone(Quaternion(1, 0, 1, 0))
+        @test !isone(Quaternion(1, 0, 0, 1))
+    end
 
-    @testset "+" begin end
+    @testset "isfinite" begin
+        @test isfinite(Quaternion(1.0, 2.0, 3.0, 4.0))
+        for inf in (Inf, -Inf)
```

Comment:
Same as `test/Octonion.jl`, this would be better with replacing with `val in (Inf, -Inf, NaN)`.